core/src/main/scala/cats/Traverse.scala

```diff
@@ -30,6 +30,12 @@ import simulacrum.typeclass
     U.TC.traverse(fa)(a => U.subst(f(a)))(this)
 
   /**
+   * A traverse followed by flattening the inner result.
+   */
+  def traverseM[G[_], A, B](fa: F[A])(f: A => G[F[B]])(implicit G: Applicative[G], F: FlatMap[F]): G[F[B]] =
+    G.map(traverse(fa)(f))(F.flatten)
```

Comment:
We got `F` (`FlatMap[F]`) as an `Applicative[F]`, but we could also use the `Traverse` itself : `G.map(traverse(fa)(f)(this))(F.flatten)`.

I am not sure if there is a reason to choose one over the other ?